src/Microsoft.AspNetCore.WebSockets.Internal/HandshakeHelpers.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.WebSockets.Internal
+{
+    public static class HandshakeHelpers
+    {
+        /// <summary>
+        /// Gets request headers needed process the handshake on the server.
+        /// </summary>
+        public static readonly IEnumerable<string> NeededHeaders = new[]
```

Comment:
Quite likely. These were brought over from aspnet/WebSockets, so I'll do a pass to check for what's needed in there.